src/dataStore/RepoDriverDataProxy.sol

```diff
@@ -0,0 +1,168 @@
+    /// Nothing happens to the currently splittable funds, but when they are split
+    /// after this function finishes, the new splits configuration will be used.
+    /// Because anybody can call `split` on `Drips`, calling this function may be frontrun
+    /// and all the currently splittable funds will be split using the old splits configuration.
+    /// @param accountId The ID of the configured account.
+    /// The caller must be the owner of the account.
+    /// @param splitsHash The hash of the list of the sender's splits receivers to be set,
+    /// the actual list must be stored in DripsDataStore.
+    /// Must be sorted by the splits receivers' addresses, deduplicated and without 0 weights.
+    /// Each splits receiver will be getting `weight / TOTAL_SPLITS_WEIGHT`
+    /// share of the funds collected by the account.
+    /// If the sum of weights of all receivers is less than `_TOTAL_SPLITS_WEIGHT`,
+    /// some funds won't be split, but they will be left for the account to collect.
+    /// It's valid to include the account's own `accountId` in the list of receivers,
+    /// but funds split to themselves return to their splittable balance and are not collectable.
+    /// This is usually unwanted, because if splitting is repeated,
+    /// funds split to themselves will be again split using the current configuration.
+    /// Splitting 100% to self effectively blocks splitting unless the configuration is updated.
+    function setSplits(uint256 accountId, bytes32 splitsHash) public whenNotPaused {
+        bytes memory data =
+            abi.encodeCall(repoDriver.setSplits, (accountId, dripsDataStore.loadSplits(splitsHash)));
+        abi.decode(_callRepoDriver(data), ());
+    }
+
+    /// @notice Emits the account metadata for the message sender.
+    /// The keys and the values are not standardized by the protocol, it's up to the users
+    /// to establish and follow conventions to ensure compatibility with the consumers.
+    /// @param accountId The ID of the emitting account.
+    /// The caller must be the owner of the account.
+    /// @param accountMetadataHash The hash of the list of account metadata,
+    /// the actual list must be stored in DripsDataStore.
+    function emitAccountMetadata(uint256 accountId, bytes32 accountMetadataHash)
+        public
+        whenNotPaused
+    {
+        bytes memory data = abi.encodeCall(
+            repoDriver.emitAccountMetadata,
+            (accountId, dripsDataStore.loadAccountMetadata(accountMetadataHash))
+        );
+        abi.decode(_callRepoDriver(data), ());
+    }
+
+    /// @notice Calls the `RepoDriver` via Caller on behalf of the `msg.sender`.
+    /// @param data The raw calldata to use when calling `RepoDriver`.
+    /// @return returnData The raw data returned from `RepoDriver`.
+    function _callRepoDriver(bytes memory data) internal returns (bytes memory returnData) {
+        return caller.callAs(_msgSender(), address(repoDriver), data);
+    }
+}
```

Comment:
Mhm, I guess you spend some time thinking about it. Can we find a more elegant solution here?

The flow currently would be:
`1. User calls Caller (signature check)`
`2. Caller calls RepoDriverDataProxy`
`3. RepoDriverDataProxy is authorized to call back Caller and calls Caller`.
`4. callAs is executed to call RepoDriver with calldata`
`5. RepoDriver using Context calls Drips`

Reply:
I can see the point, why we can't make this contract the public entry point because we would have to verify some signatures. 

Did you consider generalizing `Caller` with a function that allows to extend `calldata` with information from a DataStore?

Maybe also a bit complex

Reply:
One other idea could be to implement the signature verification part as a libary and you are using it here as well.

Reply:
We would basically allow the signature of the `calldata` with the hash to be replaced with the actual `calldata`

Reply:
The data proxies **are** the public entry points. The user needs to `approve` the data proxy address once, but after that they can call the data proxy directly, even in multiple separate transactions. The actual flow is:
1. User calls Caller.approve(RepoDriverDataProxy)

(then)

1. User calls RepoDriverDataProxy
2. RepoDriverDataProxy is authorized to call Caller and calls Caller.
3. callAs is executed to call RepoDriver with calldata
4. RepoDriver using Context calls Drips

Reply:
Mhm, how would you prevent an attacker calling for another user's account after the approval?

Reply:
`setStreams` would also allow to withdraw the funds. :thinking:

Reply:
> We would basically allow the signature of the `calldata` with the hash to be replaced with the actual `calldata`

I was thinking about storing the entire calldata and not just the configuration lists, but it seems less useful. It's more opaque, so it's more difficult to analyze if it's a part of the proposal. At least the storage streams and splits hashes match the hashes used and emitted by Drips, which too helps with transparency. The Drips API relies a lot on sending the same data multiple times, e.g. you send you splits receivers when setting splits, and then on each splitting, it's similar with streams, you send them at least twice.

Reply:
> Mhm, how would you prevent an attacker calling for another user's account after the approval?

In `_callRepoDriver` helper we have this line: `return caller.callAs(_msgSender(), address(repoDriver), data);`, which feeds the proper sender into `Caller`. If the sender is an EOA, then Caller will try to act on behalf of that EOA. If the sender is `Caller`, then the same sender will be looped back to `Caller`.

Reply:
I see. Then it make sense since only the `_msgSender()` is passed